packages/dart/lib/src/telemetry_processing/telemetry_buffer.dart

```diff
@@ -23,20 +25,92 @@ class BufferedItem<T> {
   BufferedItem(this.item, this.encoded);
 }
 
-/// In-memory buffer with time and size-based flushing.
+/// In-memory buffer with time, item count and size-based flushing.
 class InMemoryTelemetryBuffer<T extends JsonEncodable>
     extends TelemetryBuffer<T> {
-  InMemoryTelemetryBuffer();
+  final SdkLogCallback _logger;
+  final EnvelopeBuilder<T> _envelopeBuilder;
+  final Transport _transport;
+  final TelemetryBufferConfig _config;
+
+  Timer? _flushTimer;
+  final List<BufferedItem<T>> _items = [];
+  int _bufferSize = 0;
+
+  InMemoryTelemetryBuffer({
+    required SdkLogCallback logger,
+    required EnvelopeBuilder<T> envelopeBuilder,
+    required Transport transport,
+    TelemetryBufferConfig config = const TelemetryBufferConfig(),
+  })  : _logger = logger,
+        _envelopeBuilder = envelopeBuilder,
+        _transport = transport,
+        _config = config;
 
   @override
   void add(T item) {
-    final encoded = utf8JsonEncoder.convert(item.toJson());
-    final _ = BufferedItem(item, encoded);
-    // TODO(next-pr): finish this impl
+    try {
+      final encoded = utf8JsonEncoder.convert(item.toJson());
+      _items.add(BufferedItem(item, encoded));
+      _bufferSize += encoded.length;
+
+      if (_bufferSize >= _config.maxBufferSizeBytes ||
```

Comment:
I'm not sure if we need to handle this on the client, but users could add single logs which are larger 3 MB. With this check, we'd still send it, and the server would then reject it. Leading to endless re-tries (currently on cocoa, this part will be fixed). Maybe worth checking how the behaviour of this case is on iOS/Android/Flutter Web.

Reply:
We've already checked this: https://github.com/getsentry/sentry-dart/issues/3356